### ISSUE TITLE
empty lines in html attributes break markdown 

### ISSUE THREAD

**Issue by milahu** [jgm/pandoc#8307]
**Explain the problem.**

```sh
cd $(mktemp -d)

cat >test.html <<'EOF'
<table>
<tr>
<td title="this

breaks">hello</td>
</tr>
</table>
EOF

pandoc -t markdown_strict test.html -o test.md
```

actual test.md

```md
<table>
<tbody>
<tr class="odd">
<td title="this

breaks">hello</td>
</tr>
</tbody>
</table>
```

syntax highlighting in vscode breaks on these empty lines

github will render the `<td>` as

```
<td><div title="a
<p data-sourcepos=" dir="auto">b"&gt;
hello<p dir="auto"></p>
</div></td>
```

expected test.md: escape extra newlines with `&#x0A;`

```md
<td title="this&#x0A;
works">hello</td>
```

**Pandoc version?**

2.17.1.1

**Comment by jgm** [comment]
I'm sorry, I don't understand the problem.

Pandoc faithfully passes through the raw HTML, since there is no other way to represent this table in strict markdown.

The HTML is valid, as far as I can see.  If you disagree, please link to a reference that explains why it isn't.

If vscode highlighting breaks on this, it's a flaw in vscode highlighting, not pandoc.

I'm not sure I understand the issue in GitHub that you mention. But let's try pasting the HTML below to see what happens:

<table>
<tbody>
<tr class="odd">
<td title="this

breaks">hello</td>
</tr>
</tbody>
</table>

**Comment by jgm** [comment]
Ah, I see the issue.  GitHub uses an extension of commonmark, which uses different conventions for dealing with raw HTML than original markdown. 

If you are targeting GitHub, you should use `-t gfm` instead of `-t markdown_strict`.  That will give you

```
|       |
|-------|
| hello |
```

Or you can use `-t commonmark` (if you don't want the pipe table extension). In this case you'll get

```
<table>
<tbody>
<tr class="odd odd">
<td title="this

breaks">hello</td>
</tr>
</tbody>
</table>
```

*This* is indeed a bug, because running `pandoc -f commonmark` on this yields something unexpected:
```
<table>
<tbody>
<tr class="odd odd">
<td title="this

<p>breaks"&gt;hello</td></p>
</tr>
</tbody>
</table>
```

So indeed, there is a bug here, in the commonmark writer.  We need to ensure that there are no blank lines inside raw HTML blocks.

**Comment by milahu** [comment]
> jgm closed this as completed in f3e9669

ideally this should preserve all empty lines
and not add spaces

so 6 `\n` in html (pilcrows ¶ added for clarity)

```html
<td title="this¶
¶
¶
¶
¶
¶
breaks">hello</td>
```

gives give 3 `\n` + 3 `&#10;` markdown

```html
<td title="this¶
&#10;¶
&#10;&#10;¶
breaks">hello</td>
```

or 2 `\n` + 4 `&#10;` markdown

```html
<td title="this¶
&#10;&#10;&#10;&#10;¶
breaks">hello</td>
```


to clarify:

> If you are targeting GitHub, you should use `-t gfm` instead of `-t markdown_strict`.

my target is github blob, for example
https://github.com/milahu/random/blob/master/markdown/test-github-markdown-renderer.md

gfm is used in github issues

**Comment by jgm** [comment]
`gfm` is used for both, I believe.  You'll see that pipe tables work there, for example.

Anyway, with the fix I pushed earlier, things look good:
```
pandoc -f html -t commonmark
<table>
<tr>
<td title="this





breaks">hello</td>
</tr>
</table>

^D
<table>
<tbody>
<tr class="odd">
<td title="this
&#10;
&#10;
&#10;breaks">hello</td>
</tr>
</tbody>
</table>
```
